File: solutions/_1512_number_of_good_pairs/solution.py
from typing import List


class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        count = [0] * 101
        result = 0
        for num in nums:
            result += count[num]
            count[num] += 1
        return result


# https://leetcode.com/problems/number-of-good-pairs/

File: solutions/_1512_number_of_good_pairs/solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function (nums) {
  const count = new Array(101).fill(0);
  let result = 0;
  for (const num of nums) {
    result += count[num]++;
  }
  return result;
};

module.exports = { numIdenticalPairs };

/*
https://leetcode.com/problems/number-of-good-pairs/
*/
